### PR TITLE
Specialist publisher end-to-end test

### PR DESCRIPTION
**Pull request by michaelihejirika**
I have added a few more scenarios to the exist specialist publisher end-to-end test. I have also made the test to run independent of each other. Please have a look. Comments are most welcome.

### CODE REVIEW

**Comment by thomasleese** [comment]
Looks good, tests pass on my machine.

One point, I understand that, as a team, we should be using double quotes for all strings (even ones without interpolation), although I don't know if this rule is written down anywhere.

**Comment by timblair** [comment]
@thomasleese: there's a [GOV.UK Ruby style guide](https://github.com/alphagov/styleguides/blob/master/ruby.md), and [alphagov/govuk-lint](https://github.com/alphagov/govuk-lint) exists to automate the relevant linting.

TL;DR: Unless there's good reason, stick to double quotes.

Also, if you ever do need to change quotes (or whitespace, etc), keep those changes in separate commits to any "logic" changes.

**Comment by michaelihejirika** [comment]
@thomasleese @timblair thanks for your feedback. I shall make the necessary corrections.

File: spec/specialist_publisher_spec.rb [hunk 1]
```diff
@@ -54,8 +54,22 @@
...
+      visit_first_published_document
+      unpublish
+      expect_unpublished
       visit_aaib_index
+      edit_first_unplublished_document
+      fill_in("Title", with: title)
+      set_minor_update
+      save_draft
+      expect_preview_draft_link
+      publishing_unpublished
```
Comment (danielroseman) [inline spec/specialist_publisher_spec.rb]: quibble but this method should probably just be "publish_unpublished".

File: spec/support/specialist_publisher_assertion_helpers.rb [hunk 2]
```diff
@@ -31,9 +31,17 @@ def expect_error(message)
...
+  def expect_preview_draft_link
+    expect(page).to have_link("Preview draft")
+  end
+
   def expect_unpublished
     expect(find(".alert").text).to match(/^Unpublished/)
   end
 
+  def expect_discarded_draft
+    expect(find(".alert").text).to match(/^Discarded/)
```
Comment (danielroseman) [inline spec/support/specialist_publisher_assertion_helpers.rb]: Would be good to have a more specific expectation here, eg verifying that the row for the document itself now has a state of "published" rather than "published with new draft", or in the case of a never-published item, that it doesn't appear in the list at all.

File: spec/specialist_publisher_spec.rb [hunk 3]
```diff
@@ -72,11 +86,56 @@
...
+      expect_title(title)
+      visit_esi_index
+      select_published_with_new_draft
+      discard_draft
+      expect_discarded_draft
+    end
+
+    scenario "Discarding unpublished document with new draft" do
+      unpublish
+      edit_unplublished_document
```
Comment (danielroseman) [inline spec/specialist_publisher_spec.rb]: typo

File: spec/support/specialist_publisher_navigation_helpers.rb [hunk 4]
```diff
@@ -7,12 +7,26 @@ def visit_aaib_create
     visit("#{Plek.new.find('specialist-publisher')}/aaib-reports/new")
   end
 
-  def set_aaib_occurence_date(date)
+  def visit_esi_index
+    visit("#{Plek.new.find('specialist-publisher')}/esi-funds")
```
Comment (danielroseman) [inline spec/support/specialist_publisher_navigation_helpers.rb]: it would be worth extracting this repeated Plek call into a separate method. Also it should just be `Plek.find`, not `Plek.new.find`.

File: spec/specialist_publisher_spec.rb [hunk 5]
```diff
@@ -54,8 +54,22 @@
       expect_title(title)
     end
 
-    scenario "Major edit" do
+    scenario "publishing - unpublished document with new draft" do
+      visit_first_published_document
+      unpublish
+      expect_unpublished
       visit_aaib_index
+      edit_first_unplublished_document
```
Comment (michaelihejirika) [inline spec/specialist_publisher_spec.rb]: @danielroseman done

File: spec/support/specialist_publisher_assertion_helpers.rb (hunk 2 above)
Comment (michaelihejirika) [inline spec/support/specialist_publisher_assertion_helpers.rb]: @danielroseman done

File: spec/support/specialist_publisher_navigation_helpers.rb (hunk 4 above)
Comment (michaelihejirika) [inline spec/support/specialist_publisher_navigation_helpers.rb]: @danielroseman please have a look
